src/dev/LCD.cpp

```diff
@@ -23,71 +39,217 @@ void LCD::commandWrite(uint8_t data) {
-                          //                                                                                                 |WHITE|
+void LCD::driveColumn(uint8_t page, uint8_t colUp, uint8_t colLow, uint8_t data) {
+    this->commandWrite(0xAE);// Display OFF
+
+    this->commandWrite(0x40);         // Line to start writing on (0 -> 64) moves set bits with it DO NOT CHANGE
+    this->commandWrite(0xB0 + page);  // Writes the page address (4 bits, 8 rows selected by values 0-7 )
+    this->commandWrite(0x10 + colUp); // Writes the first 4 bits of the column select (out of 8 bits)
+    this->commandWrite(0x00 + colLow);// Writes the second 4 bits of the column select (out)
+
+    this->dataWrite(data);
+
+    this->commandWrite(0xAF);// Finish Writing
+    /*
+     * writes 8 vertical bits based on value between 0-255 based on bits set ex: 01001100(0x4C) is
+     * |WHITE|
+     * |BLACK|
+     * |WHITE|
+     * |WHITE|
+     * |BLACK|
+     * |BLACK|
+     * |WHITE|
+     * |WHITE|
+     */
 }
 
-void LCD::clearLCD(const uint8_t* bitMap) {
-    uint8_t page = 0xB0;
-    this->commandWrite(0xAE);          //Display OFF
-    this->commandWrite(0x40);          //Display start address + 0x40
-    for (int i = 0; i < 8; i++) {      //64 pixel display / 8 pixels per page = 8 pages
-        this->commandWrite(page);      //send page address
-        this->commandWrite(0x10);      //column address upper 4 bits + 0x10
-        this->commandWrite(0x00);      //column address lower 4 bits + 0x00
-        for (int j = 0; j < 128; j++) {//128 columns wide
-            this->dataWrite(0x00);     //write clear pixels
-            bitMap++;
+void LCD::clearLCD() {
+    unsigned char page = 0xB0;// The starting page
+
+    this->commandWrite(0xAE);                      // Display OFF
+    this->commandWrite(0x40);                      // Display start address + 0x40
+    for (uint8_t i = 0; i < 8; i++) {              // 64 pixel display / 8 pixels per page = 8 pages
+        this->commandWrite(page);                  // Send page address
+        this->commandWrite(0x10);                  // Column address upper 4 bits + 0x10
+        this->commandWrite(0x00);                  // Column address lower 4 bits + 0x00
+        for (uint8_t j = 0; j < screenSizeX; j++) {// 128 columns wide
+            this->dataWrite(0x00);                 // Write clear pixels
         }
-        page++;//after 128 columns, go to next page
+        page++;// After 128 columns, go to next page
     }
     this->commandWrite(0xAF);
 }
 
+void LCD::clearArea(uint8_t width, uint8_t height, uint8_t page, uint8_t column) {
+    // Calculate the correct column address from an 8-bit number.
+    // The issue is that the display takes 4 bit numbers, instead of 8.
+    // So you need to split it into two 4 bit numbers.
+    uint8_t columnUpperAddress = column;
```

Comment:
This can be done a bit more simply by using bitwise AND. For example, you can get the upper address with `uint8_t columnUpperAddress = (column & 0xf0) >> 4;`